src/VisualStudio/Core/Def/Library/ObjectBrowser/ObjectList.cs

```diff
@@ -747,8 +747,12 @@ protected override async Task GoToSourceAsync(uint index, VSOBJGOTOSRCTYPE srcTy
         {
             try
             {
-                using var token = _manager.AsynchronousOperationListener.BeginAsyncOperation(nameof(GoToSourceAsync));
-                using var context = _manager.OperationExecutor.BeginExecute(ServicesVSResources.IntelliSense, EditorFeaturesResources.Navigating, allowCancellation: true, showProgress: false);
+
+                var asynchronousOperationListener = _manager.ComponentModel.GetService<IAsynchronousOperationListenerProvider>().GetListener(FeatureAttribute.LibraryManager);
```

Comment:
📝 There are multiple locations in this feature that are incorrectly handling asynchronous operations. We shouldn't rely on each location to obtain the correct specific listener on demand.